nestkernel/event.h

```diff
@@ -746,63 +746,32 @@ class SecondaryEvent : public Event
   virtual fwit& operator>>( fwit& pos ) = 0;
 };
 
-/**
- * Iterator class for the GapJEvent data array.
- * Generating a copy of the array at several times is too time consuming,
- * therefore we use iterators pointing to the "old" memory location of the array
- */
-class CoeffArrayIterator
+template < typename T >
+size_t size_uint_t( T )
```

Comment:
How about:
```c++
template<typename T>
size_t sizeof_uint<T>() {...}
```

which can then just be called with:
```c++
size_uint_t<T>()
```

so it's clear to users & compilers that it's a type constant?
In fact, I bet if it's done a bit more obscurely with a ?: operator, it turns into a compile time constant (not a practical suggestion).